### ISSUE TITLE
[rstream] Meta streams

### ISSUE THREAD

**Issue by allforabit** [thi-ng/umbrella#6]
Is it possible to get subscriptions to generate new streams via transducers? I can see there's a 'resolve' subscription class to handle promises but I don't see an equivalent for streams. In other words, the final transducer would generate new streams which would be merged (using something similar to 'resolve') and could be subscribed to. This seems to be a common pattern with Rx and is done using the mergemap operator.

**Comment by postspectacular** [comment]
If I understand you correctly (and reading up on `mergeMap`) you could do this with the current state of things like this, but it's not great and I shall package this up more nicely (though I find the name `mergeMap` highly confusing).

```
results = new rs.StreamMerge([]);
results.subscribe(rs.trace());

rs.fromIterable(["hello", "world"]).subscribe(tx.map(x => results.add(rs.fromIterable([x, x, x]))));

// hello
// hello
// hello
// world
// world
// world
// done
```

Testing this I also just realized  that it would make sense for `StreamMerge` to have an option to not automatically go into DONE state when the last (possibly only) input stream is done. Currently you can't use `fromIterableSync` (the synchronous version of `fromIterable`) with `StreamMerge`, since once the first stream created by the transducer is done, no further additions to the merge are allowed... will add a separate issue!

**Comment by allforabit** [comment]
Yes this does exactly what I was looking for. I suppose you could have a dummy stream to keep it open as an alternative to adding the option to infinitely keep the stream open. Thank you!

**Comment by postspectacular** [comment]
No need for a dummy stream. StreamMerge internally creates subscriptions, but currently self-destructs (goes into final DONE state) and then won't allow new inputs to be added. I already added the option, but need to do some more related refactoring...

**Comment by postspectacular** [comment]
@allforabit just FYI, now there's another option to create these:

https://github.com/thi-ng/umbrella/blob/master/packages/rstream/README.md#meta-streams
